src/System.Linq/src/System/Linq/Repeat.cs

```diff
@@ -79,10 +79,7 @@ public TResult[] ToArray()
                 TResult[] array = new TResult[_count];
                 if (_current != null)
```

Comment:
@JonHanna That is not particularly efficient though because structs will be boxed and a call will be made. By contrast `_current != null` has non-existent overhead for value types because it's eliminated by the JIT.